tests/integration/tests/config_validations_test.go

```diff
@@ -63,11 +66,29 @@ func TestServiceEntryLabelsNotMatch(t *testing.T) {
 
 	// the DR with error, labels not match with SE
 	name := "dest-rule-labels-wrong"
-	config, _, err := utils.IstioConfigDetails(utils.BOOKINFO, name, kubernetes.DestinationRules)
+	config, err := getConfigDetails(utils.BOOKINFO, name, kubernetes.DestinationRules, assert)
 	assert.Nil(err)
 	assert.NotNil(config)
 	assert.False(config.IstioValidation.Valid)
 	assert.NotEmpty(config.IstioValidation.Checks)
 	assert.Len(config.IstioValidation.Checks, 1)
 	assert.Equal("This subset's labels are not found in any matching host", config.IstioValidation.Checks[0].Message)
 }
+
+func getConfigDetails(namespace, name, configType string, assert *assert.Assertions) (*models.IstioConfigDetails, error) {
+	config, _, err := utils.IstioConfigDetails(namespace, name, configType)
+	if err == nil && config != nil && config.IstioValidation != nil {
+		return config, nil
+	}
+	pollErr := wait.Poll(time.Second, time.Minute, func() (bool, error) {
+		config, _, err := utils.IstioConfigDetails(namespace, name, configType)
+		assert.Nil(err)
+		assert.NotNil(config)
```

Comment:
I think including these here would still cause the test to fail even if the polling returned successfully eventually. If we got a 404 back then `err != nil`. We want to poll until the object exists right?